src/plugins/ui_core_views/cell_evaluation/evaluation_plugin.ts

```diff
@@ -372,6 +371,23 @@ export class EvaluationPlugin extends CoreViewPlugin {
       }
       exportedSheetData.cells[xc] = content;
       exportedSheetData.cellValues[xc] = value;
+      const spillZone = this.getSpreadZone(position);
+      if (spillZone) {
+        exportedSheetData.arrayFormulaCells[xc] = this.getters.getRangeString(
+          this.getters.getRangeFromZone(this.getters.getActiveSheetId(), spillZone),
+          this.getters.getActiveSheetId()
```

Comment:
The position might not be on the active sheet
```suggestion
          position.sheetId
```